### PR TITLE
Allow to wait for a given status after creating a compute environment and launching a workflow.

### PR DESCRIPTION
**Pull request by jordeu**
# Description

Added new `--wait` flag that waits until a given status before exit. 

Added to commands:
- `tw compute-envs add ...`
- `tw compute-envs import ...`
- `tw launch ...`

Exit code:
- It return with a 0 exit code if the entity reaches the desired status. 
- It returns an error 1 exit code if the entity reaches a status that is incompatible with the desired status. (for example if a workflow desired status is 'RUNNING' and the job gets status 'FAILED' or 'CANCELLED' before).

### CODE REVIEW

**Comment by pditommaso** [comment]
How the desired status is specified?

**Comment by ewels** [comment]
Looks like this should also close https://github.com/seqeralabs/tower-cli/issues/177 👏🏻 🚀

**Comment by jordeu** [comment]
> How the desired status is specified?

Basically `--wait=<STATUS>`. I've updated the description with an usage example.

**Review by pditommaso — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -144,6 +144,8 @@ Pipelines consist of a pipeline repository, launch parameters, and a Compute Env
 tw launch my_sleepy_pipeline
 ```
 
+> Add a `--wait=SUCCEEDED` if you want the command to wait until the pipeline execution is complete.
```
Comment (ewels) [inline README.md:147]: Might be nice to have some mention of the different exit codes as well (and under which conditions they occur) 👍🏻

Comment (pditommaso) [reply README.md:147]: I think the error code are only 0 (ok) and 1 (fail)

Comment (ewels) [reply README.md:147]: Yup, but that's useful to know! 😉 

I was also slightly unclear on the nitty gritty. For example, if I do `--wait=FAILED` and the run finishes with `SUCCEEDED` will `tw` effectively hang, waiting forever? Or exit with `1`? Or exit with `0` because it was a successful run?

Comment (ewels) [reply README.md:147]:
```suggestion
Add a `--wait=SUCCEEDED` if you want the command to wait until the pipeline execution is complete.

When using `--wait`, `tw` can exit with one of two exit codes:

- `0`: When the run/compute-env reaches the desired state.
- `1`:  When the run/compute-env reaches a state that makes it impossible to reach the desired state.
```

Comment (jordeu) [reply README.md:147]: hang waiting forever (until you press Ctrl+C) :laughing:

Comment (jordeu) [reply README.md:147]: Ok, let me fix these corner cases. I'm going to return a 1 exit code, because I imagine that someone does a `--wait=FAILED` because he wants to test that the run fails (like a unit testing of a pipeline).
